hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -219,6 +227,89 @@ public synchronized void stop() {
     }
   }
 
+  /** A stream of {@link ContainerInfo} for supporting circular iterations. */
```

Comment:
If we limit the inflight replications, I don't think we need this change any longer to limit the number of containers we process. We should always just process all containers and then skip adding replications there are no space for in the queue.